### PR TITLE
Add glossary terms to chapter content

### PR DESCRIPTION
**Pull request by eAlasdair**
Adds a bunch of glossary terms for words I thought were important to have glossary terms for.
Only focusing on terms marked in bold in #1017 and those I came up with as I read through the content

It is by no means an exhaustive addition of glossary terms or uses for them

Also fixes some formatting issues I found along the way

### CODE REVIEW

**Comment by JackMorganNZ** [comment]
Shall we remove this from 3.0.0 milestone, or merge this now and do smaller chunks over time?

**Comment by eAlasdair** [comment]
> Shall we remove this from 3.0.0 milestone, or merge this now and do smaller chunks over time?

I definitely think some terms I added are needed for 3.0. So I'm fine allowing this to be reviewed as it stands now

File: csfieldguide/chapters/content/en/data-representation/sections/images-and-colours.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ Because a colour is simply made up of amounts of the primary colours (red, green
 
 # Pixel
 
-The word **pixel** is short for "picture element".
+The word *{glosasry-link term="pixel"}pixel{glossary-link end}* is short for "picture element".
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/data-representation/sections/images-and-colours.md]: This doesn't get rendered correctly.

File: csfieldguide/chapters/content/en/data-representation/sections/numbers.md [hunk 2]
```diff
@@ -256,7 +256,7 @@ When we have 8 bits the binary numbers start to get useful — they can represen
 
 # What is a byte?
 
-Groups of 8 bits are so useful that they have their own name: a **byte**.
+Groups of 8 bits are so useful that they have their own name: a *{glossary-term link="byte"}byte{glossary-link end}*.
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/data-representation/sections/numbers.md]: This doesn't get rendered correctly.

File: csfieldguide/chapters/content/en/data-representation/sections/text.md [hunk 3]
```diff
@@ -110,7 +110,7 @@ Nowadays ASCII has been supplanted by a code called "UTF-8", which happens to be
 
 Have a go at the following ASCII exercises:
 
-- How would you represent "science" in ASCII?
+- How would you represent "science" in ASCII (ignore the `"` marks)?
 - How would you represent "Wellington" in ASCII? (note that it starts with an upper case "W")
 - How would you represent "358" in ASCII (it is three characters, even though it looks like a number)
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/data-representation/sections/text.md]: This needs a question mark and technically the one below it as well. Also maybe we should put the part inside parentheses after the question mark on the first bullet point to be consistent.

File: csfieldguide/chapters/content/en/human-computer-interaction/sections/usability-heuristics.md [hunk 4]
```diff
@@ -5,24 +5,24 @@
 # HCI posters for your classroom!
 
 A set of posters that support this section have been published by Jennifer Gottschalk.
-[These are available in PDF format here]().
+These are available in PDF format here:
 
 {button-link link="files/HCI-posters.pdf" file="yes" text="Download HCI Posters"}
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/human-computer-interaction/sections/usability-heuristics.md]: Not related to your changes but the fact that this has a scroll bar is strange. I think the container for the button is wider than the panel.

image

File: csfieldguide/chapters/content/en/human-computer-interaction/sections/usability-heuristics.md [hunk 5]
```diff
@@ -5,24 +5,24 @@
...
 
 {panel end}
 
-Evaluating an interface is best done by getting feedback from having lots of potential users try it out.
+Evaluating an {glossary-link term="interface"}interface{glossary-link end} is best done by getting feedback from having lots of potential users try it out.
 However, this can be expensive and time-consuming, so HCI experts have come up with some quick rules of thumb that help us spot obvious problems quickly.
-The formal word for a rule of thumb is a *heuristic*, and in this section we will look at some common heuristics that can be used to critique an interface.
+The formal word for a rule of thumb is a *{glossary-link term="heuristic"}heuristic{glossary-link end}*, and in this section we will look at some common heuristics that can be used to critique an interface.
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/human-computer-interaction/sections/usability-heuristics.md]: Do you think the definition for heuristic is too long? Or do you think all of that information is needed?

File: csfieldguide/chapters/content/en/introduction/sections/programming.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # Programming
 
-And what about programming? You can get through this whole guide without doing any programming, although we'll suggest exercises.
+And what about {glossary-link term="program"}programming{glossary-link end}? You can get through this whole guide without doing any programming, although we'll suggest exercises.
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/introduction/sections/programming.md]: New sentence new line?

File: csfieldguide/chapters/content/en/data-representation/sections/images-and-colours.md (hunk 1 above)
Comment (eAlasdair) [reply csfieldguide/chapters/content/en/data-representation/sections/images-and-colours.md]: glosasry, oops

File: csfieldguide/chapters/content/en/data-representation/sections/numbers.md (hunk 2 above)
Comment (eAlasdair) [reply csfieldguide/chapters/content/en/data-representation/sections/numbers.md]: glossary-term link... wow

File: csfieldguide/chapters/content/en/data-representation/sections/text.md [hunk 7]
```diff
@@ -110,10 +110,10 @@ Nowadays ASCII has been supplanted by a code called "UTF-8", which happens to be
 
 Have a go at the following ASCII exercises:
 
-- How would you represent "science" in ASCII (ignore the `"` marks)?
+- How would you represent "science" in ASCII? (ignore the `"` marks)
 - How would you represent "Wellington" in ASCII? (note that it starts with an upper case "W")
-- How would you represent "358" in ASCII (it is three characters, even though it looks like a number)
-- How would you represent "Hello, how are you?" (look for the comma, question mark, and space characters in ASCII table)
+- How would you represent "358" in ASCII? (it is three characters, even though it looks like a number)
+- How would you represent "Hello, how are you?" in ASCII?(look for the comma, question mark, and space characters in ASCII table)
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/data-representation/sections/text.md]: Need a space between the question mark and parentheses

**Pull request merged**